sycl/source/detail/graph_impl.hpp

```diff
@@ -899,7 +899,7 @@ class graph_impl {
   void addEventForNode(std::shared_ptr<graph_impl> GraphImpl,
                        std::shared_ptr<sycl::detail::event_impl> EventImpl,
                        std::shared_ptr<node_impl> NodeImpl) {
-    if (!(EventImpl->getCommandGraph()))
+    if ((nullptr != EventImpl) && !(EventImpl->getCommandGraph()))
```

Comment:
```suggestion
    if (EventImpl && !(EventImpl->getCommandGraph()))
```

Can simplify this and also remove the yoda notation.